Samba 4.16.5

DavidePrincipi opened the pull request:
Refs https://github.com/NethServer/dev/issues/6702

DavidePrincipi committed:
Samba 4.16.5

DavidePrincipi committed:
Fix upgrade with missing Copr RPMs

Install gnutls 3 library from Copr repo

https://download.copr.fedorainfracloud.org/results/sergiomb/SambaAD

DavidePrincipi committed:
fixup! Fix upgrade with missing Copr RPMs

DavidePrincipi committed:
fixup! Fix upgrade with missing Copr RPMs

DavidePrincipi committed:
fixup! Fix upgrade with missing Copr RPMs

DavidePrincipi commented:
Samba services start, but if I run

    ldapsearch -ZZ -H ldap://<IP of DC> -D '' -w '' -b '' -s base
    
then I get in `journalctl -M nsdc`:

    Sep 28 23:13:11 nsdc-nscom1.ad.dp.nethserver.net samba[27]: _tstream_tls_accept_send: TLS ../../source4/lib/tls/tls_tstream.c:1300 - The request is invalid.. Failed to set default priorities

DavidePrincipi commented:
We should add a gnutls.config file to nethserver-dc RPM

See https://lists.samba.org/archive/samba/2020-December/233651.html


Thanks to @gsanchietti

DavidePrincipi committed:
fixup! Fix upgrade with missing Copr RPMs

DavidePrincipi commented:
TLS is working now. This is a diff of available ciphers

```diff
@@ -3,67 +3,46 @@
 | ssl-enum-ciphers: 
 |   TLSv1.0: 
 |     ciphers: 
-|       TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_3DES_EDE_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_RC4_128_MD5 - strong
-|       TLS_RSA_WITH_RC4_128_SHA - strong
 |     compressors: 
 |       NULL
 |   TLSv1.1: 
 |     ciphers: 
-|       TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_3DES_EDE_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_RC4_128_MD5 - strong
-|       TLS_RSA_WITH_RC4_128_SHA - strong
 |     compressors: 
 |       NULL
 |   TLSv1.2: 
 |     ciphers: 
-|       TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_AES_128_CBC_SHA256 - strong
+|       TLS_DHE_RSA_WITH_AES_128_CCM - strong
 |       TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 - strong
 |       TLS_DHE_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 - strong
+|       TLS_DHE_RSA_WITH_AES_256_CCM - strong
 |       TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA256 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_GCM_SHA256 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA256 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_GCM_SHA384 - strong
-|       TLS_RSA_WITH_3DES_EDE_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 - strong
 |       TLS_RSA_WITH_AES_128_CBC_SHA - strong
-|       TLS_RSA_WITH_AES_128_CBC_SHA256 - strong
+|       TLS_RSA_WITH_AES_128_CCM - strong
 |       TLS_RSA_WITH_AES_128_GCM_SHA256 - strong
 |       TLS_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_RSA_WITH_AES_256_CBC_SHA256 - strong
+|       TLS_RSA_WITH_AES_256_CCM - strong
 |       TLS_RSA_WITH_AES_256_GCM_SHA384 - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA256 - strong
-|       TLS_RSA_WITH_CAMELLIA_128_GCM_SHA256 - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA256 - strong
-|       TLS_RSA_WITH_CAMELLIA_256_GCM_SHA384 - strong
-|       TLS_RSA_WITH_RC4_128_MD5 - strong
-|       TLS_RSA_WITH_RC4_128_SHA - strong
 |     compressors: 
 |       NULL
 |_  least strength: strong
```

DavidePrincipi commented:
In new domain provisioning procedure:

    Sep 29 11:22:36 nsdc-nscom1.ad.dp.nethserver.net samba-tool[30]: ERROR(ldb): uncaught exception - 0000052D: Constraint violation - check_password_restrictions: the password does not meet the complexity criteria!

gsanchietti commented on root/etc/e-smith/events/actions/nethserver-dc-install:
```diff
@@ -37,7 +37,7 @@ if [[ ! -x ${nsroot}/usr/sbin/samba ]]; then
     cp -f /etc/yum/vars/* ${nsroot}/etc/yum/vars
     rpm --root=${nsroot} --import /etc/pki/rpm-gpg/*
-    yum -y --releasever=/ --installroot=${nsroot} install /usr/lib/nethserver-dc/ns-samba-*.rpm centos-release systemd-networkd bind-utils ntp | /usr/libexec/nethserver/ptrack-nsdc-install
+    yum -y --releasever=/ --installroot=${nsroot} install /usr/lib/nethserver-dc/compat-*.rpm /usr/lib/nethserver-dc/gmp-6.1.2-1.el7.x86_64.rpm /usr/lib/nethserver-dc/ns-samba-*.rpm centos-release systemd-networkd bind-utils ntp | /usr/libexec/nethserver/ptrack-nsdc-install
```
We could avoid hardcoding the release:
```suggestion
    yum -y --releasever=/ --installroot=${nsroot} install /usr/lib/nethserver-dc/compat-*.rpm /usr/lib/nethserver-dc/gmp-*.rpm /usr/lib/nethserver-dc/ns-samba-*.rpm centos-release systemd-networkd bind-utils ntp | /usr/libexec/nethserver/ptrack-nsdc-install
```

gsanchietti commented on root/etc/e-smith/events/actions/nethserver-dc-upgrade:
```diff
@@ -33,7 +33,13 @@ rsync -ai --delete /etc/yum.repos.d/ ${nsroot}/etc/yum.repos.d
-yum -y --releasever=/ --installroot=${nsroot} update \* /usr/lib/nethserver-dc/ns-samba-*.rpm -y | /usr/libexec/nethserver/ptrack-nsdc-install
+{
+    echo 'install' /usr/lib/nethserver-dc/compat-*.rpm
+    echo 'install' /usr/lib/nethserver-dc/gmp-6.1.2-1.el7.x86_64.rpm
```
```suggestion
    echo 'install' /usr/lib/nethserver-dc/gmp-*.rpm
```

DavidePrincipi committed:
Apply suggestions from code review

Co-authored-by: Giacomo Sanchietti <giacomo.sanchietti@nethesis.it>

DavidePrincipi committed:
Fix domain provisioning procedure

The provision fail because Perl interpreter is not installed but it is
required by checkpassword.pl

Refs 97654c87f42359273a00c11483fb84606940dc4b

DavidePrincipi committed:
Fix nsdc-run-worker for Python 3.6

The container has now the Python 3.6 interpreter.
